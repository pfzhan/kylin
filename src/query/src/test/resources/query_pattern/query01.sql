--
-- Licensed to the Apache Software Foundation (ASF) under one
-- or more contributor license agreements.  See the NOTICE file
-- distributed with this work for additional information
-- regarding copyright ownership.  The ASF licenses this file
-- to you under the Apache License, Version 2.0 (the
-- "License"); you may not use this file except in compliance
-- with the License.  You may obtain a copy of the License at
--
--     http://www.apache.org/licenses/LICENSE-2.0
--
-- Unless required by applicable law or agreed to in writing, software
-- distributed under the License is distributed on an "AS IS" BASIS,
-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-- See the License for the specific language governing permissions and
-- limitations under the License.
--
select count (distinct o_year) as distinct_o_year, peru_volumn / volume + sqrt(4) as mkt_share
from (
select year(o_orderdate) as o_year,
l_extendedprice * (1 - l_discount) + 2 as volume,
case when n2.n_name = 'PERU' then l_extendedprice * (1 - l_discount)
else 0 end as peru_volumn
from tpch.lineitem
inner join tpch.part on l_partkey = p_partkey
inner join tpch.supplier on l_suppkey = s_suppkey
inner join tpch.orders on l_orderkey = o_orderkey
inner join tpch.customer on o_custkey = c_custkey
inner join tpch.nation n1 on c_nationkey = n1.n_nationkey
inner join tpch.nation n2 on s_nationkey = n2.n_nationkey
inner join tpch.region on n1.n_regionkey = r_regionkey
where r_name = 'AMERICA'
and o_orderdate between '1995-01-01' and '1996-12-31'
and p_type = 'ECONOMY BURNISHED NICKEL'
) as all_nations
group by peru_volumn / volume + sqrt(4)
order by peru_volumn / volume + sqrt(4)