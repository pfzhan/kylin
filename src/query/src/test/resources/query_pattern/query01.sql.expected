SELECT COUNT(DISTINCT "O_YEAR") "DISTINCT_O_YEAR", (("PERU_VOLUMN" / "VOLUME") + SQRT(4)) "MKT_SHARE"
FROM (SELECT (YEAR("O_ORDERDATE")) "O_YEAR", (("L_EXTENDEDPRICE" * (1 - "L_DISCOUNT")) + 2) "VOLUME", (CASE WHEN ("N2"."N_NAME" = 'PERU') THEN ("L_EXTENDEDPRICE" * (1 - "L_DISCOUNT")) ELSE 0 END) "PERU_VOLUMN"
FROM "TPCH"."LINEITEM"
INNER JOIN "TPCH"."PART" ON ("L_PARTKEY" = "P_PARTKEY")
INNER JOIN "TPCH"."SUPPLIER" ON ("L_SUPPKEY" = "S_SUPPKEY")
INNER JOIN "TPCH"."ORDERS" ON ("L_ORDERKEY" = "O_ORDERKEY")
INNER JOIN "TPCH"."CUSTOMER" ON ("O_CUSTKEY" = "C_CUSTKEY")
INNER JOIN "TPCH"."NATION" "N1" ON ("C_NATIONKEY" = "N1"."N_NATIONKEY")
INNER JOIN "TPCH"."NATION" "N2" ON ("S_NATIONKEY" = "N2"."N_NATIONKEY")
INNER JOIN "TPCH"."REGION" ON ("N1"."N_REGIONKEY" = "R_REGIONKEY")
WHERE ((("R_NAME" = 'A') AND ("O_ORDERDATE" BETWEEN '2010-01-01' AND '2010-01-01')) AND ("P_TYPE" = 'A'))) "ALL_NATIONS"
GROUP BY (("PERU_VOLUMN" / "VOLUME") + SQRT(4))
ORDER BY (("PERU_VOLUMN" / "VOLUME") + SQRT(4))